DesignSafe-CI/portal
# Task/des 3047 create custom query hooks

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [DES-3047](https://tacc-main.atlassian.net/browse/DES-3047)

## Summary of Changes: ##
- create a hooks for fetching queries
- add :get_events hook
- add: get_eventTypes hook
- add: get_opentopo hook - preprocessing data from api to form center points
- add: endpoints for opentopo data in Django - views.py & urls.py
- config change in docker-compose-dev.all.debug.yml to increase the cache limit for memcached server
## Testing Steps: ##
1.  make sure that - https://designsafe.dev/recon-portal/opentopo - fetches the data-
2. after initial query- the above /opentopo endpoint should fetch data instantly.
3. console log the - useGetEvents();  useGetEventTypes(); useGetOpenTopo();
4. ensure the results match, what the Angular version queries returned.

## UI Photos:

## Notes: ##
- Make sure you change the docker-compose-debug.all.yml based on the laptop processor and run the make start - to get the cache to work.
